packages/plugins/src/foundry/decode.ts

```diff
@@ -144,25 +143,12 @@ export const makeNetworkConfig = (
     gasEstimates,
   } = deploymentInfo
 
-  // Each Merkle leaf must have a gas amount that's at most 80% of the block gas limit. This ensures
-  // that it's possible to execute the transaction on-chain. Specifically, there must be enough gas
-  // to execute the Sphinx Module's logic, which isn't included in the gas estimate of the Merkle
-  // leaf. The 80% was chosen arbitrarily.
-  const maxAllowedGasPerLeaf = (BigInt(8) * BigInt(blockGasLimit)) / BigInt(10)
-
   const parsedActionInputs: Array<ActionInput> = []
   const unlabeledContracts: NetworkConfig['unlabeledContracts'] = []
   for (let i = 0; i < accountAccesses.length; i++) {
     const { root, nested } = accountAccesses[i]
     const gas = gasEstimates[i].toString()
 
-    if (BigInt(gas) > maxAllowedGasPerLeaf) {
-      const networkName = fetchNameForNetwork(BigInt(chainId))
-      throw new Error(
-        `Estimated gas for a transaction is too close to the block gas limit on ${networkName}.`
-      )
-    }
-
```

Comment:
This change is irrelevant to the PR, but I'm removing it anyways because this check is pointless and confusing. If a leaf is too large, the `findMaxBatchSize` function in the simulation will throw an error. Keeping this here will only confuse maintainers in the future. I'm removing it now because I noticed that you mentioned you were modifying this value when working on Rootstock

Reply:
I'm not like super opinionated on this, but the UX of this error definitely seems better than the error we throw in the simulation. So, I would lean towards keeping this. If we do, we should use the getMaxGasLimit function to fetch the limit instead of doing the calculation here.

Reply:
Wdym "the UX of this error"? Do you mean that it's better for us to throw this error before the simulation, or do you mean that this error message is better? If it's the latter, i'd rather just improve the error message in the simulation

Reply:
I'm good with keeping it. I improved the error message by using the `decodedAction`, which required me to put it further down in this function